kafka/src/kafka_client.rs

```diff
@@ -0,0 +1,74 @@
+use std::time::Duration;
+
+use rdkafka::{
+    config::FromClientConfig,
+    consumer::{Consumer, DefaultConsumerContext, MessageStream, StreamConsumer},
+    producer::{future_producer::OwnedDeliveryResult, FutureProducer, FutureRecord},
+    util::Timeout,
+    ClientConfig,
+};
+
+pub struct KafkaClient {
+    client_config: ClientConfig,
+    timeout: Timeout,
+}
+
+impl KafkaClient {
+    pub fn new(
+        bootstrap_server: &str,
+        group_id: Option<&str>,
+        batch_interval: Option<&str>,
+    ) -> Result<Self, Box<dyn std::error::Error>> {
+        let mut client_config: ClientConfig = ClientConfig::new();
+
+        client_config
+            .set("bootstrap.servers", bootstrap_server)
+            .set("group.id", group_id.unwrap_or("1"))
+            .set("queue.buffering.max.ms", batch_interval.unwrap_or("0"));
+
+        Ok(KafkaClient {
+            client_config,
+            timeout: Timeout::After(Duration::ZERO),
+        })
+    }
+
+    pub fn create_producer(&self) -> Result<FutureProducer, Box<dyn std::error::Error>> {
+        let producer: FutureProducer = FutureProducer::from_config(&self.client_config)?;
+        Ok(producer)
```

Comment:
```suggestion
        FutureProducer::from_config(&self.client_config)
```
will this work? same with `create_consumer`